kitsune/lib/email.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+
+log = logging.getLogger('k.lib.email')
+RealBackend = import_string(settings.EMAIL_LOGGING_REAL_BACKEND)
+
+
+class LoggingEmailBackend(RealBackend):
+    """
+    Wraps a email backend defined in Django's settings and logs everything it does.
+    """
+
+    def open(self):
+        """Open a network connection."""
+        new_conn = super(LoggingEmailBackend, self).open()
+        # Checking for True or False directly instead of relying on trutheyness
+        # to avoid catching non-booleans
+        if new_conn is True:
+            log.debug('Succesfully opened new connection.')
+        elif new_conn is False:
+            log.debug('Did not open a new connection. (Either cached or failed)')
+        else:
+            assert (
+                new_conn is None,
+                'Unexpected return from email backend open method: %r.' % (new_conn, )
+            )
```

Comment:
If possible, I'd like to avoid explicitly checking what the backend is. Like the other assertion, this is about validating expectations. If this assert fails, then the backend didn't act like I expected it to, so things should blow up.

I tried to use assertions to validate my assumptions, and non-assertion code to deal with what I expected the backends to do.